base/compiler/abstractinterpretation.jl

```diff
@@ -1589,6 +1589,16 @@ function invoke_rewrite(xs::Vector{Any})
     return newxs
 end
 
+function abstract_add_finalizer(interp::AbstractInterpreter, argtypes::Vector{Any}, sv::InferenceState)
+    if length(argtypes) == 3
+        tt = argtypes[3]
+        finalizer_argvec = Any[argtypes[3], argtypes[2]]
+        call = abstract_call(interp, ArgInfo(nothing, finalizer_argvec), sv, 1)
+        return CallMeta(Nothing, Effects(), FinalizerInfo(call.info, call.effects))
```

Comment:
Semantically this is currently incorrect since our runtime current agree to run the latest version of the finalizer as of the execution time (which is unknowable), not as of compilation time. It feels hard to imagine a scenario where this would matter to someone however. Do we just want to update the documentation to make this execution-world behavior be undefined (stated as choosing "the parent world, or any later one.")?

I assume the inliner should add an edge here too, so it mostly shouldn't be a problem visible to Revise-users either.

Reply:
Yes, I'd like to just update the documentation to make that statement. In theory we could codegen an inline world switch, but I don't think it's worth it.

Reply:
Documentation was updated.